### PR TITLE
Change codePoint and code point to codepoint

### PR DESCRIPTION
**Pull request by dmk42**
We used three different spellings for the same thing: codepoint, codePoint, and code point.  This change normalizes our usage to codepoint.

1.  Changed Chapel local variables named `codepoint` to `cp` to avoid confusion with the method of that name.
2.  Changed `[Cc]odePoint` to `[Cc]odepoint`.
3.  Added deprecation for the `codePointIndex` type and `codePointToString()`.
4.  Changed ``code point`` to ``codepoint`` in English text.

Requires a change to #12561 

- [x] Tested `types/string`, `release/examples`, and all changed tests on Linux and Mac
- [x] Tested `make docs`
- [x] Full GASNet test to make sure multilocale is OK

### CODE REVIEW

**Comment by dmk42** [comment]
@ronawho - Thanks!

File: modules/internal/String.chpl [hunk 1]
```diff
@@ -218,62 +218,70 @@ module String {
...
+  record codepointIndex {
     pragma "no doc"
     var _cpindex  : int;
   }
 
-  // Helper routines in support of being able to use ranges of codePointIndex
+  /*
+   Deprecated, use `codepointIndex`.
+  */
+  proc codePointIndex type {
```
Comment (mppf) [inline modules/internal/String.chpl]: Normally we `pragma "no doc"` deprecated types/functions

File: modules/internal/String.chpl [hunk 2]
```diff
@@ -484,19 +492,19 @@ module String {
...
       var n = 0;
-      for codepoint in this.codePoints() do
+      for cp in this.codepoints() do
         n += 1;
       return n;
     }
 
     /*
-      Deprecated, use :proc:`string.numCodePoints`.
+      Deprecated, use :proc:`string.numCodepoints`.
```
Comment (mppf) [inline modules/internal/String.chpl]: Again, normally we "no doc" such things

File: modules/internal/String.chpl [hunk 3]
```diff
@@ -599,28 +607,28 @@ module String {
...
-        yield codepoint;
+        qio_decode_char_buf(cp, nbytes, multibytes, maxbytes);
+        yield cp;
         i += nbytes;
       }
     }
 
     /*
-      Deprecated, use :proc:`string.codePoints`.
+      Deprecated, use :proc:`string.codepoints`.
```
Comment (mppf) [inline modules/internal/String.chpl]: And here.

File: modules/internal/String.chpl [hunk 4]
```diff
@@ -2197,6 +2205,14 @@ module String {
     return s;
   }
 
+  /*
+    Deprecated, use :proc:`codepointToString`.
+  */
+  inline proc codePointToString(i: int(32)) {
```
Comment (mppf) [inline modules/internal/String.chpl]: and here

**Comment by mppf** [comment]
@dmk42 - thanks, I'm happy with this (post-merge) but we might consider no-doc'ing the deprecated routines. That could happen in this release, or not.

File: modules/internal/String.chpl (hunk 1 above)
Comment (ben-albrecht) [reply modules/internal/String.chpl]: I don't think this is true. When this was last discussed, it was determined that we would continue to document deprecated features. This was documented in the [deprecation guidelines](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/Deprecation.rst):

> Continue to document and support the deprecated feature for 1 release cycle - The documentation should contain a pointer to the preferred feature - This holds true for modules as well

Comment (mppf) [reply modules/internal/String.chpl]: Ah, then I must be doing it wrong...

**Pull request merged**